src/schema/v2/me/myBids.ts

```diff
@@ -0,0 +1,259 @@
+    _args,
+    {
+      causalityLoader,
+      meLoader,
+      saleArtworksLoader,
+      saleArtworksAllLoader,
+      salesLoaderWithHeaders,
+      saleLoader,
+    }
+  ) => {
+    if (
+      !(
+        causalityLoader &&
+        meLoader &&
+        saleArtworksLoader &&
+        saleArtworksAllLoader &&
+        salesLoaderWithHeaders &&
+        saleLoader
+      )
+    ) {
+      return null
+    }
+
+    /**
+     * Causality's connection is faux with a max cap of 25, so we'll only ever
+     * be able to return a non-pagable single list. Increased to an arbitrarily
+     * high number to account for registered sales and watched lots.
+     */
+    const FETCH_COUNT = 99
+
+    // Grab userID to pass to causality
+    const me = await meLoader()
+
+    // Fetch all auction lot standings from a given user
+    const causalityPromise = causalityLoader({
+      query: gql`
+        query LotStandingsConnection($userId: ID!, $first: Int) {
+          lotStandingConnection(userId: $userId, first: $first) {
+            edges {
+              node {
+                isHighestBidder
+                lot {
+                  bidCount
+                  internalID
+                  reserveStatus
+                  saleId
+                  soldStatus
+                  floorSellingPriceCents
+                  sellingPriceCents
+                  onlineAskingPriceCents
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        first: FETCH_COUNT,
+        userId: me.id,
+      },
+    })
+
+    // Queue up promise for all registered sales
+    const registeredSalesPromise = salesLoaderWithHeaders({
+      registered: true,
+      is_auction: true,
+      size: FETCH_COUNT,
+    })
+
+    // Queue up promise for watched lots
+    const watchedSaleArtworksPromise = saleArtworksAllLoader({
+      include_watched_artworks: true,
+      total_count: true,
+      first: FETCH_COUNT,
+    })
+
+    // Fetch everything in parallel
+    const [
+      causalityResponse,
+      registeredSalesResponse,
+      watchedSaleArtworksResponse,
+    ] = await Promise.all([
+      causalityPromise,
+      registeredSalesPromise,
+      watchedSaleArtworksPromise,
+    ])
+
+    // Map over response to gather all sale IDs
+    const causalityLots = causalityResponse.lotStandingConnection.edges.map(
+      ({ node }) => node.lot
+    )
+    const causalitySaleIds = causalityLots.map((lot) => lot.saleId)
+    const registeredSaleIds = registeredSalesResponse.body.map(
+      (sale) => sale._id
+    )
+    const watchedSaleIds = watchedSaleArtworksResponse.body.map(
+      (artwork) => artwork.sale_id
+    )
+
+    // Combine ids from categories and dedupe
+    const combinedSaleIds = _.uniq([
+      ...causalitySaleIds,
+      ...registeredSaleIds,
+      ...watchedSaleIds,
+    ])
+
+    // Fetch all sales to format into a list
+    let combinedSales = await Promise.all(combinedSaleIds.map(saleLoader))
```

Comment:
non-blocking: `.map(loader)` is a bit hard to parse vs `map(id => loader(id))` - i would probably do it too though.